contrib/cf-remote/cf_remote/commands.py

```diff
@@ -136,9 +135,18 @@ def download(tags=None, version=None, edition=None):
         print("No suitable packages found")
     else:
         for artifact in artifacts:
-            download_package(artifact.url)
+            if download:
+                download_package(artifact.url)
+            else:
+                print(artifact.url)
```

Comment:
I'd prefer `yield artifact.url` here and then doing the expected thing with the url in `list()`/`download()`.

Reply:
I started changing this, and then realized that I disagree, in this case. The function does some printing before iterating over the packages, and if it was going to return a list / generator of packages, I'd prefer if it was more _pure_, and the printing was left out. Then the printing would have to be duplicated in both commands, or split into another function, and the iteration function would be pretty much pointless, it could be replaced with a generator expression, and then we are back to square 0.